compress.go

```
@@ -0,0 +1,211 @@
+		payload := packets[:payloadLen]
+		uncompressedLen := payloadLen
+
+		buf.Reset()
+		buf.Write(blankHeader) // Buffer.Write() never returns error
+
+		// If payload is less than minCompressLength, don't compress.
+		if uncompressedLen < minCompressLength {
+			buf.Write(payload)
+			uncompressedLen = 0
+		} else {
+			zCompress(payload, buf)
```

golangci-lint (1.62.2)

Error return value is not checked

(errcheck)